src/display/editor/outliner.js

```diff
@@ -317,6 +343,452 @@ class HighlightOutline extends Outline {
   get box() {
     return this.#box;
   }
+
+  get free() {
+    return false;
+  }
+}
+
+class FreeOutliner {
+  #box;
+
+  #bottom = [];
+
+  #innerMargin;
+
+  #top = [];
+
+  // The first 6 elements are the last 3 points of the top part of the outline.
+  // The next 6 elements are the last 3 points of the line.
+  // The next 6 elements are the last 3 points of the bottom part of the
+  // outline.
+  // We track the last 3 points in order to be able to:
+  //  - compute the normal of the line,
+  //  - compute the control points of the quadratic Bézier curve.
+  #last = new Float64Array(18);
+
+  #min;
+
+  #min_dist;
+
+  #scaleFactor;
+
+  #thickness;
+
+  #points = [];
+
+  static #MIN_DIST = 8;
+
+  static #MIN_DIFF = 2;
+
+  static #MIN = FreeOutliner.#MIN_DIST + FreeOutliner.#MIN_DIFF;
+
+  constructor({ x, y }, box, scaleFactor, thickness, innerMargin = 0) {
+    this.#box = box;
+    this.#thickness = thickness * scaleFactor;
+    this.#last.set([NaN, NaN, NaN, NaN, x, y], 6);
+    this.#innerMargin = innerMargin;
+    this.#min_dist = FreeOutliner.#MIN_DIST * scaleFactor;
+    this.#min = FreeOutliner.#MIN * scaleFactor;
+    this.#scaleFactor = scaleFactor;
+    this.#points.push(x, y);
+  }
+
+  get free() {
+    return true;
+  }
+
+  isEmpty() {
+    return isNaN(this.#last[8]);
```

Comment:
Please add a comment about why we check that particular element, since that's not immediately obvious (at least to me)?